NitroxClient/GameLogic/Cyclops.cs

```diff
@@ -53,5 +75,452 @@ public void ActivateShield(string guid)
+            cyclops.gameObject.RequireComponent<SubFire>().liveMixin.health = subFireHealth;
+
+            // unusedDamagePoints is checked against damagePoints to determine if there's enough damage points. Failing to set the new list
+            // of unusedDamagePoints will cause random DamagePoints to appear.
+            damageManager.ReflectionSet("unusedDamagePoints", unusedDamagePoints);
+            // Visual update only to show the water leaking through the window and various hull points based on missing health.
+            damageManager.ReflectionCall("ToggleLeakPointsBasedOnDamage", false, false, null);
+        }
+
+        /// <summary>
+        /// A new fire is a state change to the Cyclops. If it's the owner, the other players need to be updated
+        /// </summary>
+        public void OnCreateFire(SubFire subFire, SubFire.RoomFire startInRoom)
+        {
+            OnTakeDamage(subFire.subRoot, null);
+        }
+
+        /// <summary>
+        /// Create a new fire in a specific room and node. Calling <see cref="SubFire.CreateFire(SubFire.RoomFire)"/> would just cause it to use 
+        /// a random number generator to choose the node. This can trigger sending <see cref="CyclopsFireHealthChanged"/>
+        /// and <see cref="CyclopsDamage"/> packets
+        /// </summary>
+        public void CreateFire(SubRoot subRoot, KeyValuePair<CyclopsRooms, SubFire.RoomFire> fireRoom, int nodeIndex)
+        {
+            Transform transform2 = fireRoom.Value.spawnNodes[nodeIndex];
+            fireRoom.Value.fireValue++;
+            PrefabSpawn component = transform2.GetComponent<PrefabSpawn>();
+            if (component == null)
+            {
+                return;
+            }
+            GameObject gameObject = component.SpawnManual();
+            Fire componentInChildren = gameObject.GetComponentInChildren<Fire>();
+            if (componentInChildren)
+            {
+                componentInChildren.fireSubRoot = subRoot;
+            }
+        }
+
+        /// <summary>
+        /// Called when <see cref="Fire.Douse(float)"/> is called. This can trigger sending <see cref="CyclopsFireHealthChanged"/>
+        /// and <see cref="CyclopsDamage"/> packets
+        /// </summary>
+        public void OnFireDoused(Fire fire, SubRoot subRoot, float douseAmount)
+        {
+            SubFire subFire = subRoot.gameObject.RequireComponent<SubFire>();
+            Dictionary<CyclopsRooms, SubFire.RoomFire> roomFiresDict = (Dictionary<CyclopsRooms, SubFire.RoomFire>)subFire.ReflectionGet("roomFires");
+            string fireGuid = GuidHelper.GetGuid(fire.gameObject);
+
+            if (!fireDouseAmount.ContainsKey(fireGuid))
+            {
+                // 500 fires in a single session? Are they trying to cook Peepers on open fires? Yes, I've tried to do it, and to my disappointment it
+                // doesn't cook them.
+                if (fireDouseAmount.Count > 500)
```

Comment:
Why is this check needed?

Reply:
Debugging code I forgot to remove. Can't remember the exact reason, but it's no longer possible for the fire dousing amount to exceed 15.